util/conditions/getter.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Getter interface defines methods that a Cluster API object should implement in order to
+// use the conditions package for getting conditions.
+type Getter interface {
+	controllerutil.Object
```

Comment:
Do we have to embed this?

Reply:
I'm embedding this because we want to be able to get access to the object metadata, which is required in a follow-up PR that implements conditions merge.
Additionally, this can be useful if in future we want to add support for ObservedGeneration

Reply:
Yes we want the objects we deal with to be Kubernetes objects

Reply:
It just makes it a wider interface than is being used right now. If you plan on using it later I'm fine to leave it in here.

Reply:
I'm fine both in leaving as it is or reducing the interface to the minimum set of required methods (so far `GetObjectKind`, `GetName`)

Reply:
@vincepri I leave to you the last call on this ^^

Reply:
Let's keep it for posterity, we might want to offer patching bits later or something similar, it might be useful